astronomer/providers/dbt/cloud/hooks/dbt.py

```diff
@@ -0,0 +1,127 @@
+                default_account_id = connection.login
+                if not default_account_id:
+                    raise AirflowException("Could not determine the dbt Cloud account.")
+                bound_args.arguments["account_id"] = int(default_account_id)
+
+        return await func(*bound_args.args, **bound_args.kwargs)
+
+    return cast(T, wrapper)
+
+
+class DbtCloudHookAsync(BaseHook):
+    """
+    Interact with dbt Cloud using the V2 API.
+
+    :param dbt_cloud_conn_id: The ID of the :ref:`dbt Cloud connection <howto/connection:dbt-cloud>`.
+    """
+
+    conn_name_attr = "dbt_cloud_conn_id"
+    default_conn_name = "dbt_cloud_default"
+    conn_type = "dbt_cloud"
+    hook_name = "dbt Cloud"
+
+    def __init__(self, dbt_cloud_conn_id: str):
+        self.dbt_cloud_conn_id = dbt_cloud_conn_id
+        self.base_url = ""
+
+    async def get_headers(self) -> Dict[str, Any]:
+        """Get Headers, base url from the connection details"""
+        headers: Dict[str, Any] = {}
+        connection: Connection = await sync_to_async(self.get_connection)(self.dbt_cloud_conn_id)
+        tenant = connection.schema if connection.schema else "cloud"
+        self.base_url = f"https://{tenant}.getdbt.com/api/v2/accounts/"
+        provider_info = get_provider_info()
+        package_name = provider_info["package-name"]
+        version = provider_info["versions"]
+        headers["User-Agent"] = f"{package_name}-v{version}"
+        headers["Content-Type"] = "application/json"
+        headers["Authorization"] = f"Token {connection.password}"
+        return headers
+
+    def get_request_url_params(
+        self, endpoint: str, include_related: Optional[List[str]] = None
+    ) -> Tuple[str, Dict[str, Any]]:
+        """Form URL from base url and endpoint url"""
+        data: Dict[str, Any] = {}
+        if include_related:
+            data = {"include_related": include_related}
+        if self.base_url and not self.base_url.endswith("/") and endpoint and not endpoint.startswith("/"):
+            url = self.base_url + "/" + endpoint
+        else:
+            url = (self.base_url or "") + (endpoint or "")
+        return url, data
+
+    @provide_account_id
+    async def get_job_details(
+        self, run_id: int, account_id: Optional[int] = None, include_related: Optional[List[str]] = None
+    ) -> Any:
+        """
+        Uses Http async call to retrieve metadata for a specific run of a dbt Cloud job.
+
+        :param run_id: The ID of a dbt Cloud job run.
+        :param account_id: Optional. The ID of a dbt Cloud account.
+        :param include_related: Optional. List of related fields to pull with the run.
+            Valid values are "trigger", "job", "repository", and "environment".
+        """
+        endpoint = f"{account_id}/runs/{run_id}/"
+        headers = await self.get_headers()
+        url, params = self.get_request_url_params(endpoint, include_related)
+        print("headers ", headers)
```

Comment:
Is this print necessary? If so, let's use `logging` instead.

Reply:
No its necessary to print it I used it for my local development will remove it